sdk/storage/inc/blobs/protocol/blob_rest_client.hpp

```diff
@@ -1098,7 +1154,222 @@ namespace Azure { namespace Storage { namespace Blobs {
             std::move(response), std::move(pHttpResponse));
       }
 
+      struct GetPropertiesOptions
+      {
+        Azure::Core::Nullable<int32_t> Timeout;
+      }; // struct GetPropertiesOptions
+
+      static Azure::Core::Response<BlobServiceProperties> GetProperties(
+          Azure::Core::Context context,
+          Azure::Core::Http::HttpPipeline& pipeline,
+          const std::string& url,
+          const GetPropertiesOptions& options)
+      {
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, url);
+        request.AddQueryParameter("restype", "service");
+        request.AddQueryParameter("comp", "properties");
+        request.AddHeader("x-ms-version", c_APIVersion);
+        if (options.Timeout.HasValue())
+        {
+          request.AddQueryParameter("timeout", std::to_string(options.Timeout.GetValue()));
```

Comment:
I thought we aggreed that timeout should be added by pipeline.

Reply:
I'll create an issue to track that

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/370